src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -0,0 +1,610 @@
+                                                                            images_folder_default_name=images_folder_name,
+                                                                            annotations_filename=annotations_filename,
+                                                                            verbose=verbose and not silent)
+        # Model initialization if needed
+        if self.model is None and self.checkpoint_load_iter != 0:
+            # No model loaded, initializing new
+            self.init_model()
+            # User set checkpoint_load_iter, so they want to load a checkpoint
+            # Try to find the checkpoint_load_iter checkpoint
+            checkpoint_name = "checkpoint_iter_" + str(self.checkpoint_load_iter) + ".pth"
+            checkpoints_folder = os.path.join(self.parent_dir, '{}_checkpoints'.format(self.experiment_name))
+            full_path = os.path.join(checkpoints_folder, checkpoint_name)
+            try:
+                checkpoint = torch.load(full_path, map_location=torch.device(self.device))
+            except FileNotFoundError as e:
+                e.strerror = "File " + checkpoint_name + " not found inside checkpoints_folder, " \
+                                                         "provided checkpoint_load_iter (" + \
+                             str(self.checkpoint_load_iter) + \
+                             ") doesn't correspond to a saved checkpoint.\nNo such file or directory."
+                raise e
+            if not silent and verbose:
+                print("Loading checkpoint:", full_path)
+
+            load_state(self.model, checkpoint)
+        elif self.model is None:
+            raise AttributeError("self.model is None. Please load a model or set checkpoint_load_iter.")
+
+        self.model = self.model.eval()  # Change model state to evaluation
+        self.model.to(self.device)
+        if "cuda" in self.device:
+            self.model = self.model.to(self.device)
+            if self.half:
+                self.model.half()
+
+        if self.multiscale:
+            self.scales = [0.5, 1.0, 1.5, 2.0]
+
+        coco_result = []
+        num_keypoints = Pose.num_kpts
+
+        pbar_eval = None
+        if not silent:
+            pbar_desc = "Evaluation progress"
+            pbar_eval = tqdm(desc=pbar_desc, total=len(data), bar_format="{l_bar}%s{bar}{r_bar}" % '\x1b[38;5;231m')
+
+        img_height = data[0]['img'].shape[0]
+
+        if img_height in (1080, 1440):
+            offset = 200
+        elif img_height == 720:
+            offset = 50
+        else:
+            offset = 0
+
+        for sample in data:
+            file_name = sample['file_name']
+            img = sample['img']
+            h, w, _ = img.shape
+            max_width = w
+            kernel = int(h / self.first_pass_height)
+            if kernel > 0:
+                pool_img = HighResolutionPoseEstimationLearner.__pooling(self, img, kernel)
```

Comment:
I think it's easier to call internal private methods like this:
```suggestion
                pool_img = self.__pooling(img, kernel)
```

If there was a specific reason i am missing for the original code, ignore this.